keep/providers/base/base_provider.py

```diff
@@ -710,3 +715,75 @@ def simulate_alert(cls) -> dict:
 class BaseTopologyProvider(BaseProvider):
     def pull_topology(self) -> list[TopologyServiceInDto]:
         raise NotImplementedError("get_topology() method not implemented")
+
+
+class BaseIncidentProvider(BaseProvider):
+    def _get_incidents(self) -> list[IncidentDto]:
+        raise NotImplementedError("_get_incidents() in not implemented")
+
+    def get_incidents(self) -> list[IncidentDto]:
+        return self._get_incidents()
+
+    @staticmethod
+    def _format_incident(
+        event: dict, provider_instance: "BaseProvider" = None
+    ) -> IncidentDto | list[IncidentDto]:
+        raise NotImplementedError("_format_incidents() not implemented")
+
+    @classmethod
+    def format_incident(
+        cls,
+        event: dict,
+        tenant_id: str | None,
+        provider_type: str | None,
+        provider_id: str | None,
+    ) -> IncidentDto | list[IncidentDto]:
+        logger = logging.getLogger(__name__)
+
+        provider_instance: BaseProvider | None = None
+        if provider_id and provider_type and tenant_id:
+            try:
+                # To prevent circular imports
+                from keep.providers.providers_factory import ProvidersFactory
+
+                provider_instance: BaseProvider = (
+                    ProvidersFactory.get_installed_provider(
+                        tenant_id, provider_id, provider_type
+                    )
+                )
+            except Exception:
+                logger.exception(
+                    "Failed loading provider instance although all parameters were given",
+                    extra={
+                        "tenant_id": tenant_id,
+                        "provider_id": provider_id,
+                        "provider_type": provider_type,
+                    },
+                )
+        logger.debug("Formatting Incident")
+        return cls._format_incident(event, provider_instance)
+
+    def setup_incident_webhook(
+        self,
+        tenant_id: str,
+        keep_api_url: str,
+        api_key: str,
+        setup_alerts: bool = True,
+    ) -> dict | None:
+        """
+        Setup a webhook for the provider.
+
+        Args:
+            tenant_id (str): _description_
+            keep_alerts_api_url (str): _description_
+            api_key (str): _description_
+            setup_alerts (bool, optional): _description_. Defaults to True.
+            keep_incidents_api_url (str, optional): _description_. Defaults to None.
+
+        Returns:
+            dict | None: If some secrets needs to be saved, return them in a dict.
+
+        Raises:
+            NotImplementedError: _description_
+        """
+        raise NotImplementedError("setup_webhook() method not implemented")
```

Comment:
args don't match and setup_webhook is wrong (copy/paste issue?)

Reply:
Yup, copy pasta error, fixed